### PR TITLE
AbuseIPDBPopulateIndicators to python3 + code conventions

### PR DESCRIPTION
**Pull request by samuelFain**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators.py [hunk 1]
```diff
@@ -2,25 +2,47 @@
...
+                "source": "AbuseIPDB",
+                "reputation": BLACKLISTED,
+                "seenNow": "true",
+            },
+        )
+
+    demisto.results("All Indicators were created successfully")
+
+
+if __name__ == "__main__":
```
Comment (tkatzir) [inline Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators.py:47]:
```suggestion
if __name__ in ('__main__', '__builtin__', 'builtins'):
```

**Review by tkatzir — COMMENTED** [review]
Looks good.
One minor comment.

File: Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators_test.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from CommonServerPython import DemistoException
+
+
+def test_get_contents():
+    from AbuseIPDBPopulateIndicators import get_contents
+
+    assert get_contents({}) is None
```
Comment (BEAdi) [inline Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators_test.py:9]: Change as we talked - use `mocker.patch('execute_command', return_value=[{'Contents': {}}])` with the return value you want to check (once with nothing, second time with a real response).
I would also send `args` with real values - you can use the default values that are in the yml file.

File: Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators.py [hunk 3]
```diff
@@ -2,25 +2,64 @@
...
+    )
+
+    if not res:
+        return None
+
+    return res[0].get("Contents", None)
+
+
+def check_ips(ips: dict):
+    """Check 'ips' dictionary validity
```
Comment (BEAdi) [inline Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators.py:30]: Is it a dictionary? I think it is a list (can be checked in the raw response of `"abuseipdb-get-blacklist"`).

File: Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators_test.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    with pytest.raises(DemistoException) as e:
+        check_ips(input)
+    assert str(e.value) == "No Indicators were created (possibly bad API key)"
+
+
+def test_check_ips(input=[True]):
+    from AbuseIPDBPopulateIndicators import check_ips
+
+    assert check_ips(input) is None
```
Comment (BEAdi) [inline Packs/AbuseDB/Scripts/AbuseIPDBPopulateIndicators/AbuseIPDBPopulateIndicators_test.py:24]: We can inline the input variable, and use test ip values (as we want to check the behavior when getting ips, and not some unrelated values).

**Review by BEAdi — COMMENTED** [review]
Looks great! Adding some notes about the tests.

File: Tests/conf.json [hunk 5]
```diff
@@ -5324,6 +5324,7 @@
         "Carbon Black Enterprise Protection V2 Test": "No credentials"
     },
     "skipped_integrations": {
+        "AbuseIPDBPopulateIndicators": "No instance",
```
Comment (BEAdi) [inline Tests/conf.json:5327]:
```suggestion
        "AbuseIPDB": "License expired",
```

**Review by BEAdi — COMMENTED** [review]
Looks great! Lets add documentation to the test functions (given, when, then).

**Review by tkatzir — APPROVED** [review]
Approved, but check other comments, made by other reviewers.

**Pull request merged**